tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -10443,6 +10443,12 @@ bool SpirvEmitter::emitEntryFunctionWrapperForRayTracing(
   auto *entryLabel = spvBuilder.createBasicBlock();
   spvBuilder.setInsertPoint(entryLabel);
 
+  // KHR_ray_tracing extension requires SPIR-V 1.4/Vulkan 1.2
+  if (!featureManager.isExtensionEnabled("SPV_NV_ray_tracing")) {
+    featureManager.requestTargetEnv(SPV_ENV_VULKAN_1_2, "Raytracing",
+                                    decl->getLocation());
+  }
```

Comment:
The better place to put this logic is in CapabilityVisitor.cpp. Right before adding the capability/extension you can add
```cpp
featureManager.requestTargetEnv(SPV_ENV_VULKAN_1_2, "Raytracing", {});
```


https://github.com/microsoft/DirectXShaderCompiler/blob/f0d4419ff52a7786a0ca9f221a6aae8e8d4d7f26/tools/clang/lib/SPIRV/CapabilityVisitor.cpp#L526-L527

Reply:
Agreed. Updated